toffy/bin_extraction.py

```diff
@@ -19,11 +22,22 @@ def extract_missing_fovs(bin_file_dir, extraction_dir, panel, extract_intensitie
     fovs = io_utils.remove_file_extensions(io_utils.list_files(bin_file_dir, substrs='.bin'))
     extracted_fovs = io_utils.list_folders(extraction_dir, substrs='fov')
 
-    # filter out moly fovs
-    moly_fovs = list_moly_fovs(bin_file_dir)
+    # filter out empty json file fovs
+    empty_fovs = check_for_empty_files(bin_file_dir, return_json_names=True, warn=True)
+    if empty_fovs:
+        fovs = list(set(fovs).difference(empty_fovs))
+
+    # check for moly fovs
+    moly_fovs = list_moly_fovs(bin_file_dir, fovs)
 
-    print("Skipping the following previously extracted FOVs: ", ", ".join(extracted_fovs))
-    print("Moly FOVs which will not be extracted: ", ", ".join(moly_fovs))
+    if extracted_fovs:
+        print("Skipping the following previously extracted FOVs: ", ", ".join(extracted_fovs))
+    if moly_fovs:
+        print("Moly FOVs which will not be extracted: ", ", ".join(moly_fovs))
+    if empty_fovs:
+        print("FOVs with empty json files which will not be extracted: ", ", ".join(empty_fovs))
+    if not extracted_fovs and not moly_fovs and not empty_fovs:
```

Comment:
This should go under the `if missing_fovs` conditional so that the user knows how many are getting extracted, even if some are skipped